torch/csrc/CudaIPCTypes.cpp

```diff
@@ -0,0 +1,156 @@
+#ifdef USE_CUDA
+#include <torch/csrc/CudaIPCTypes.h>
+#include <map>
+#include <mutex>
+
+namespace torch {
+
+void BadIPCPatternError() {
+  static bool warned = false;
+  if (not warned) {
+    LOG(WARNING)
+        << "Producer process has been terminated before all shared CUDA tensors released.";
+    warned = true;
+  }
+}
+
+struct CudaIPCGlobalEntities {
```

Comment:
This is my obligatory "docs please" comment; I know you told me you're still working on docs, so I won't mention it again for the rest of this review :)